apps/api/src/modules/notifications/notifications.service.ts

```diff
@@ -124,6 +139,60 @@ export class NotificationsService extends CoreService<Notification> {
     this.isProcessingQueue = false;
   }
 
+  async getProviderConfirmation(): Promise<void> {
+    this.logger.log(
+      'Starting CRON job to add notifications to queue for confirmation from provider',
+    );
+
+    if (this.isProcessingConfirmationQueue) {
+      this.logger.log(
+        'Notifications are already being added to confirmation queue, skipping this CRON job',
+      );
+      return;
+    }
+
+    this.isProcessingConfirmationQueue = true;
+    let allAwaitingConfirmationNotifications: Notification[] = [];
+
+    try {
+      allAwaitingConfirmationNotifications = await this.getAwaitingConfirmationNotifications();
+    } catch (error) {
+      this.isProcessingConfirmationQueue = false;
+      this.logger.error('Error fetching awaiting confirmation notifications');
+      this.logger.error(JSON.stringify(error, null, 2));
+      return;
+    }
+
+    this.logger.log(
+      `Adding ${allAwaitingConfirmationNotifications.length} awaiting confirmation notifications to queue`,
+    );
+
+    for (const notification of allAwaitingConfirmationNotifications) {
+      try {
+        if (notification.retryCount <= this.maxRetryCount) {
+          notification.deliveryStatus = DeliveryStatus.QUEUED_CONFIRMATION;
```

Comment:
We don't really need QUEUED Confirmation as its a temporary state and just use the QUEUED state to let know that its queued for processing. But okay to have to get some difference.

Reply:
Kept QUEUED_CONFIRMATION status only for now. No changes made here.